### PR TITLE
449 use the python installer for linux installation

### PR DESCRIPTION
**Pull request by thomas-senechal**

### CODE REVIEW

**Comment by gregLibert** [comment]
@SlnPons There is no replacement of the existing script + an update of the installation procedure because I would like to start testing it before.
And by testing I mean checking that it passes the tests that I asked @Pierregvx to do.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
           files: thyra-server_${{ matrix.target }}_${{ matrix.arch }}
 
   build-installer:
-    if: ${{ (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/tags/v') }}
+#    if: ${{ (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/tags/v') }}
```
[the remark this answers is not in the document]
Comment (gregLibert) [reply .github/workflows/build.yml:80]: okidoki. Done

File: scripts/installer.py [hunk 2]
```diff
@@ -177,18 +183,19 @@ def installThyraApp(self):
...
             subprocess.Popen([thyra_app_path], start_new_session=True)
             logging.info("Thyra App will now start. You can right click on the tray icon to start Thyra.")
 
-
     """
     Installs thyra server, thyra app and a DNS server.
     """
     def startInstall(self):
         logging.info("Starting installation of thyra")
+        self.createConfigFolder()
```
[the remark this answers is not in the document]
Comment (gregLibert) [reply scripts/installer.py:198]: my bad, sorry.

**Comment by thomas-senechal** [comment]
Didn't tested it and I can't approve it since I'm the one that made this PR. 

You have my approval.

File: scripts/installer.py [hunk 3]
```diff
@@ -144,6 +144,12 @@ def generateCACertificate(self):
             self.printErrorAndExit(f"Error while deleting mkcert binary: {err}")
         logging.info("CA certificate successfully generated")
 
+    def _moveFile(self, file, destination):
+        shutil.move(file, destination)
+    
+    def _deleteFile(self, file):
+        os.remove(thyra_server_path)
```
Comment (Thykof) [inline scripts/installer.py:151]: warning, `file` parameter is not used.

don't we have a linter that can spot that before pushing or in the CI?

File: scripts/linux_installer.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        if stdout and "installed" in stdout:
+            self.executeCommand("sudo apt-get install -y libnss3-tools", True)
+
+        super().generateCACertificate()
+    
+    def _moveFile(self, file, destination):
+        self.executeCommand(f"sudo mv {file} {destination}", True)
+
+    def _deleteFile(self, file):
+        self.executeCommand(f"sudo rm {file}", True)
```
Comment (Thykof) [inline scripts/linux_installer.py:79]: I think here you can use the methods defined in base classe Installer

File: scripts/linux_installer.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        self.MKCERT_FILENAME = "mkcert"
+
+        self.THYRA_SERVER_URL = "https://github.com/massalabs/thyra/releases/latest/download/thyra-server_linux_amd64"
+        self.THYRA_APP_URL = "https://github.com/massalabs/Thyra-Menu-Bar-App/releases/latest/download/ThyraApp_linux-amd64"
+        self.MKCERT_URL = "https://dl.filippo.io/mkcert/latest?for=linux/amd64"
+
+    def configureNetworkManager(self):
+        logging.info("Configuring NetworkManager...")
+        self.executeCommand("sudo cp /etc/NetworkManager/NetworkManager.conf /etc/NetworkManager/NetworkManager.conf.backup_thyra_install", True)
+        stdout, _stderr = self.executeCommand("grep '^dns=' /etc/NetworkManager/NetworkManager.conf", True, allow_failure=True)
```
Comment (Thykof) [inline scripts/linux_installer.py:22]: if you want, you can name this unsused variable `_`. but I understand if you want to be clear about what this variable is and that we deliberately don't want to use it. (this remark apply for some other cases in the code)

```suggestion
        stdout, _ = self.executeCommand("grep '^dns=' /etc/NetworkManager/NetworkManager.conf", True, allow_failure=True)
```

File: scripts/linux_installer.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            self.configureDNSMasq()
+        elif runningDNS == "dnsmasq":
+            logging.info("dnsmasq is already installed")
+            self.configureDNSMasq()
+        elif runningDNS == "systemd-r":
+            logging.warning("Your computer is using systemd-resolved for DNS. Thyra needs dnsmasq to redirect .massa domains to localhost.")
+            user_answer = input("Do you want to install dnsmasq and configure it to redirect .massa domains to localhost? [y/n] ")
+            user_answer = user_answer.lower()
+            if user_answer == "y" or user_answer == "yes":
+                self.configureNetworkManager()
```
Comment (Thykof) [inline scripts/linux_installer.py:61]: in that case maybe we should make sure that dndmasq is installed?

```suggestion
                self.executeCommand("sudo apt-get install -y dnsmasq", True)
                self.configureNetworkManager()
```

**Review by Thykof — CHANGES_REQUESTED** [review]
Some python and logical remarks, some are optional :) about installation I can't really say if it's OK

File: scripts/linux_installer.py (hunk 4 above)
Comment (gregLibert) [reply scripts/linux_installer.py:79]: No, the command needs sudo...

**Review by Thykof — APPROVED** [review]

**Comment by SlnPons** [comment]
@Pierregvx can you test this issue by making sure that it passes the end-to-end test that you wrote for linux please?

**Pull request merged**